detect_secrets/core/baseline.py

```diff
@@ -190,42 +184,21 @@ def merge_results(old_results, new_results):
 
     :rtype: dict
     """
-    for filename, secrets in old_results.items():
+    for filename, old_secrets in old_results.items():
         if filename not in new_results:
-            new_results[filename] = secrets
             continue
 
-        if len(secrets) == len(new_results[filename]):
-            # Assuming that secrets remain in order.
```

Comment:
Secrets are still ordered, because of https://github.com/Yelp/detect-secrets/blob/master/detect_secrets/core/baseline.py#L233. So we may not need to use this O(n^2) approach.

Reply:
Amazing catch, the solution would have been clear if it was `merge 234 and 156`, as you could easily tell 1 < 2 and increment the list with the lower element, but since we were talking `hashed_secrets`'s and I didn't want to do anything based off line numbers, I made it a `dict` instead `old_secrets_mapping[old_secret['hashed_secret']] = old_secret` 👍